src/scanner.c

```diff
@@ -1211,6 +1211,12 @@ static bool inner_scan(void *payload, TSLexer *lexer, const bool *valid_symbols)
                 } else if (BRACE_EXPR && BRACE_TYPE) {
 
                     lex_advance(lexer);
+
+                    // Start of a macro expression
+                    if (lexer->lookahead == '{') {
+                        return false;
+                    }
```

Comment:
Every branch of the overall if statement either does `lex_advance` or returns false. I think this should be extracted up to line 1114